packages/arb-avm-cpp/cmachine/machine.go

```diff
@@ -114,9 +114,43 @@ func (m *Machine) PrintState() {
 	C.machinePrint(m.c)
 }
 
+func (m *Machine) ExecuteNormalAssertion(
+	maxSteps uint64,
+	inbox value.TupleValue,
+	maxWallTime time.Duration,
+) (*protocol.ExecutionAssertion, uint64) {
+	var buf bytes.Buffer
+	_ = value.MarshalValue(inbox, &buf)
+
+	msgData := buf.Bytes()
+	msgDataC := C.CBytes(msgData)
+	defer C.free(msgDataC)
+
+	assertion := C.executeNormalAssertion(
+		m.c,
+		C.uint64_t(maxSteps),
+		msgDataC,
+		C.uint64_t(uint64(maxWallTime.Seconds())),
+	)
+
+	outMessagesRaw := toByteSlice(assertion.outMessages)
+	logsRaw := toByteSlice(assertion.logs)
+
+	return protocol.NewExecutionAssertion(
```

Comment:
I think you can refactor this so there's a single raw assertion to assertion function that this and `ExecuteAssertion` call like you did on the cavm side